### PR TITLE
[feat] adds referral codes

### LINKED ISSUE #98: [feature][large] Affiliate Codes
Let's get the reposting going, want to add affiliate codes so people can help spread the word. Making a full proposal in Quip

### PR DESCRIPTION
**Pull request by jzeiders**
Basic implementation
Closes #98

### CODE REVIEW

**Comment by jzeiders** [comment]
Also not sure how we want to do database migrations in prod
Closes #100

File: api/database/migrations/20191111203354-add-promo-codes.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"use strict";
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return Promise.all([
+      queryInterface
+        .addColumn("HackerProfiles", "promoCode", {
+          type: Sequelize.STRING(100)
```
Comment (WilhelmWillie) [inline api/database/migrations/20191111203354-add-promo-codes.js]: we want promoCode to be unique right? prob want to add the unique attribute to this field https://stackoverflow.com/questions/28116187/unique-constraint-on-sequelize-column

Comment (jzeiders) [reply api/database/migrations/20191111203354-add-promo-codes.js]: true, will fail in the off chance that someone generates a duplicate but should be fine...

**Comment by jzeiders** [comment]
What's our process for doing db migrations to production?

**Comment by WilhelmWillie** [comment]
yeah so rn, we'll have to manually migrate the production DB by running

`NODE_ENV=production npx sequelize-cli db:migrate`

I guess a next step right now in terms of dev-ex is to auto deploy + migrate on merges to our main branch which is why proper reviews and etc. will be more more important when doing DB migrations 

For now, I say: document and verify the intended behavior is working as expected on a dev DB, approve + merge PR, run prod migration, then deploy to prod Heroku -- will spend some time tonight and tomorrow cleaning up this experience

**Pull request merged**